### DISCUSSION TITLE
issue with AudioWAVServer

### DISCUSSION THREAD

**Discussion by OMENGT** [pschatzmann/arduino-audio-tools#954] [Q&A] [answered]
I've tried a few of the examples using AudioWAVServer and made a few of my own, it works using vlc and ffmpeg but if i connect and disconnect it fails to reconnect again, I can see on the serial port that StreamCopier.h is active but the stream does not restart. I get this error in ffmpeg: **Invalid data found when processing input**

I tried the AudioWAVServerEx examples but got this compiler error which I don't understand:

In file included from C:\Users\TH~1.POI\AppData\Local\Temp\arduino_modified_sketch_995405\streams-generator-webserverex_wav1.ino:12:
C:\Users\th\Work Folders\Documents\Arduino\libraries\arduino-audio-tools-main\src/AudioLibs/AudioServerEx.h: In member function 'virtual tinyhttp::Str* audio_tools::AudioWAVServerEx::getReplyHeader()':
C:\Users\th\Work Folders\Documents\Arduino\libraries\arduino-audio-tools-main\src/AudioLibs/AudioServerEx.h:168:25: error: 'WAV_FORMAT_PCM' was not declared in this scope
             wi.format = WAV_FORMAT_PCM;
                         ^~~~~~~~~~~~~~
C:\Users\th\Work Folders\Documents\Arduino\libraries\arduino-audio-tools-main\src/AudioLibs/AudioServerEx.h:168:25: note: suggested alternative: 'LOG_FORMAT'
             wi.format = WAV_FORMAT_PCM;
                         ^~~~~~~~~~~~~~
                         LOG_FORMAT
C:\Users\th\Work Folders\Documents\Arduino\libraries\arduino-audio-tools-main\src/AudioLibs/AudioServerEx.h:174:17: error: 'class audio_tools::WAVEncoder' has no member named 'writeHeader'
             enc.writeHeader(&mp);
                 ^~~~~~~~~~~
Multiple libraries were found for "WiFi.h"
 Used: C:\Users\th\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.11\libraries\WiFi
 Not used: C:\apps\arduino-1.8.19\libraries\WiFi
Multiple libraries were found for "SD.h"
 Used: C:\Users\th\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.11\libraries\SD
 Not used: C:\apps\arduino-1.8.19\libraries\SD
exit status 1
Error compiling for board ESP32 Dev Module.

Thanks in advance for any help

**Comment by pschatzmann** [comment, marked as answer]
Difficult to say what you did wrong, but WAV_FORMAT_PCM does not exist any more. It has been replaced by the enum values of the enum class AudioFormat.

I can't reproduce your issue with the examples of the latest version of the library. 

Most likley to tried to mix different versions.

**Reply by OMENGT** [reply]
Thanks, I updated libraries to latest version and AudioWAVServerEx compiles and works now

Still have the problem with AudioWAVServer not reconnecting, but I'm AudioWAVServerEx now so it doesn't really matter

Thanks very much
